sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/Samplers.java

```diff
@@ -147,14 +147,47 @@ public static Sampler alwaysOff() {
    * whether or not to sample. If there is no parent, the Sampler uses the provided Sampler delegate
    * to determine the sampling decision.
    *
-   * @param delegateSampler the {@code Sampler} which is used to make the sampling decisions if the
-   *     parent does not exist.
+   * @param root the {@code Sampler} which is used to make the sampling decisions if the parent does
+   *     not exist.
    * @return a {@code Sampler} that follows the parent's sampling decision if one exists, otherwise
-   *     following the delegate sampler's decision.
+   *     following the root sampler's decision.
    * @since 0.7.0
    */
-  public static Sampler parentOrElse(Sampler delegateSampler) {
-    return new ParentOrElse(delegateSampler);
+  public static Sampler parentBased(Sampler root) {
+    return new ParentBased(root);
+  }
+
+  /**
+   * Returns a {@link Sampler} that always makes the same decision as the parent {@link Span} to
+   * whether or not to sample. If there is no parent, the Sampler uses the provided Sampler delegate
+   * to determine the sampling decision.
+   *
+   * @param root the required {@code Sampler} which is used to make the sampling decisions if the
+   *     parent does not exist.
+   * @param remoteParentSampled the optional {@code Sampler} which is used to make the sampling
+   *     decisions if the remote parent exist and is sampled.
+   * @param remoteParentNotSampled the optional {@code Sampler} which is used to make the sampling
+   *     decisions if the remote parent exist and is not sampled.
+   * @param localParentSampled the optional {@code Sampler} which is used to make the sampling
+   *     decisions if the local parent exist and is sampled.
+   * @param localParentNotSampled the optional {@code Sampler} which is used to make the sampling
+   *     decisions if the local parent exist and is not sampled.
+   * @return a {@code Sampler} that follows the parent's sampling decision if one exists, otherwise
+   *     following the root sampler and other optional sampler's decision.
+   * @since 0.8.0
+   */
+  public static Sampler parentBased(
```

Comment:
I would use a `Builder` pattern:
```
public static Builder newBuilder(Sampler root)
```

Where Builder allows to set any other `Sampler`. this way user don't have to configure all the others default samplers.

Reply:
👍 what about `newParentBasedBuilder` ?

Reply:
seems ok for a start, yes